gax-httpjson/src/main/java/com/google/api/gax/httpjson/InstantiatingHttpJsonChannelProvider.java

```diff
@@ -196,11 +187,20 @@ private TransportChannel createChannel() throws IOException, GeneralSecurityExce
     ManagedHttpJsonChannel channel =
         ManagedHttpJsonChannel.newBuilder()
             .setEndpoint(endpoint)
-            .setDefaultHeaders(HttpJsonMetadata.newBuilder().setHeaders(headers).build())
             .setExecutor(executor)
             .setHttpTransport(httpTransportToUse)
             .build();
 
+    HttpJsonClientInterceptor headerInterceptor =
+        new HttpJsonHeaderInterceptor(headerProvider.getHeaders());
+
+    channel = new ManagedHttpJsonInterceptorChannel(channel, headerInterceptor);
+    if (interceptorProvider != null && interceptorProvider.getInterceptors() != null) {
+      for (HttpJsonClientInterceptor interceptor : interceptorProvider.getInterceptors()) {
+        channel = new ManagedHttpJsonInterceptorChannel(channel, interceptor);
```

Comment:
Are you ignoring all interceptors except the last one here?

Reply:
No. Note, `channel` is used in both left and right parts of the assignment expression - this is how the interceptors form a chain.